firebase-firestore/src/main/java/com/google/firebase/firestore/local/SQLiteIndexManager.java

```diff
@@ -147,117 +169,51 @@ public void addToCollectionParentIndex(ResourcePath collectionPath) {
   public void addFieldIndex(FieldIndex index) {
     hardAssert(started, "IndexManager not started");
 
-    int nextIndexId = memoizedMaxId + 1;
+    int nextIndexId = memoizedMaxIndexId + 1;
     index =
         FieldIndex.create(
-            nextIndexId, index.getCollectionGroup(), index.getSegments(), index.getUpdateTime());
+            nextIndexId, index.getCollectionGroup(), index.getSegments(), index.getIndexState());
 
     db.execute(
         "INSERT INTO index_configuration ("
             + "index_id, "
             + "collection_group, "
-            + "index_proto, "
-            + "update_time_seconds, "
-            + "update_time_nanos) VALUES(?, ?, ?, ?, ?)",
+            + "index_proto) VALUES(?, ?, ?)",
         nextIndexId,
         index.getCollectionGroup(),
-        encodeSegments(index),
-        index.getUpdateTime().getTimestamp().getSeconds(),
-        index.getUpdateTime().getTimestamp().getNanoseconds());
-
-    // TODO(indexing): Use a 0-counter rather than the timestamp.
-    setCollectionGroupUpdateTime(index.getCollectionGroup(), SnapshotVersion.NONE.getTimestamp());
-
+        encodeSegments(index));
```

Comment:
When we add a new field index, we will want to create an index_state entry right away if it doesn't exist. Otherwise, the index only lives in the memoized memory table and not the persistence table. We currently fetch the index configurations and merge it with the stored index_states, but could we simplify the logic if we wrote new field indexes to the index_state table right away?